File: C++/source/Framework/Messaging/LightHouse/LightHouseTransportBroadcastLightPulseMessage.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightHouseTransportBroadcastLightPulseMessage.h"
#include "Framework/Messaging/LightHouse/LightHouseTransportObjectManager.h"
#include "Framework/Messaging/LightHouse/LightHouseTransportTypes.h"
#include "Framework/Messaging/LightHouse/LightPulse.h"

namespace WaveNs
{

LightHouseTransportBroadcastLightPulseMessage::LightHouseTransportBroadcastLightPulseMessage (LightPulse *pLightPulse)
    : WaveMessage (LightHouseTransportObjectManager::getWaveServiceId (), LIGHT_HOUSE_TRANSPORT_BROADCAST_LIGHT_PULSE),
      m_pLightPulse (pLightPulse)
{
}

LightHouseTransportBroadcastLightPulseMessage::~LightHouseTransportBroadcastLightPulseMessage ()
{
    if (NULL != m_pLightPulse)
    {
        delete m_pLightPulse;
    }
}

LightPulse *LightHouseTransportBroadcastLightPulseMessage::getPLightPulse ()
{
    return (m_pLightPulse);
}

void LightHouseTransportBroadcastLightPulseMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization ();
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightHouseTransportBroadcastLightPulseMessage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

import com.CxWave.Wave.Framework.Messaging.Local.WaveMessage;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.NonSerializable;

@NonSerializable
public class LightHouseTransportBroadcastLightPulseMessage extends WaveMessage
{
    private LightPulse m_lightPulse;

    public LightHouseTransportBroadcastLightPulseMessage ()
    {
        super (LightHouseTransportObjectManager.getWaveServiceId (), LightHouseTransportTypes.LIGHT_HOUSE_TRANSPORT_BROADCAST_LIGHT_PULSE);
    }

    public LightHouseTransportBroadcastLightPulseMessage (final LightPulse lightPulse)
    {
        super (LightHouseTransportObjectManager.getWaveServiceId (), LightHouseTransportTypes.LIGHT_HOUSE_TRANSPORT_BROADCAST_LIGHT_PULSE);

        m_lightPulse = lightPulse;
    }

    public LightPulse getLightPulse ()
    {
        return m_lightPulse;
    }

    public void setLightPulse (final LightPulse lightPulse)
    {
        m_lightPulse = lightPulse;
    }
}
